x-pack/plugins/alerting/server/application/rule/methods/bulk_edit/bulk_edit_rules.ts

```diff
@@ -505,7 +505,7 @@ async function updateRuleAttributesAndParamsInMemory<Params extends RuleParams>(
     validateScheduleInterval(context, updatedRule.schedule.interval, ruleType.id, rule.id);
 
     const { modifiedParams: ruleParams, isParamsUpdateSkipped } = paramsModifier
-      ? await paramsModifier(updatedRule.params)
+      ? await paramsModifier(updatedRule as Rule<Params>)
```

Comment:
I'm not sure how to implement that without type casting. There are several similar Rule types that aren't assignable to each other due to subtle differences.

Reply:
It's fine to cast here since we're in the process of migrating and consolidating our rule types. But I would just put a TODO comment here